### DISCUSSION TITLE
Performance: JavaScript vs Go

### DISCUSSION THREAD

**Discussion by ilteoood** [pocketbase/pocketbase#3634] [Q&A] [answered]
In my actual project, I'm extending PocketBase capabilities with JS.
Considering it is a project which relies a lot on the Realtime API, what do you suggest to have the best performances: switch to go, or stay with JS?

**Comment by ganigeorgiev** [comment, marked as answer]
In general the Go code will be always more performant, but if your JS code uses primarily the exposed PocketBase JS methods, then the difference will be more or less negligible because under the hood these methods will invoke directly the Go code (see also [Go vs JS route execution](https://github.com/pocketbase/benchmarks/blob/master/results/hetzner_cax11.md#go-vs-js-route-execution) and [Go vs JS hooks execution](https://github.com/pocketbase/benchmarks/blob/master/results/hetzner_cax11.md#go-vs-js-hooks-execution); _the benchmark doesn't show it, but the JS memory consumption is slightly larger compared to the raw Go code_).

**Reply by ilteoood** [reply]
Ok, thank you so much. I thought that the object serialization overhead has higher 🤔
